src/chem/Chain.js

```diff
@@ -36,6 +46,38 @@ class Chain {
     return this._residues;
   }
 
+  getType() {
+    return this._type;
+  }
+
+  _determineType() {
+    const residues = this._residues;
+
+    for (let i = 0, n = residues.length; i < n; ++i) {
+      if ((residues[i]._type.flags & ResidueType.Flags.NUCLEIC) !== 0) {
+        this._type = CHAIN_TYPE.NUCLEIC;
+        break;
+      } else if ((residues[i]._type.flags & ResidueType.Flags.PROTEIN) !== 0) {
+        this._type = CHAIN_TYPE.AMINO;
+        break;
+      }
+    }
+
+    for (let i = 0, n = residues.length; i < n; ++i) {
+      if (residues[i]._type.flags === ResidueType.Flags.UNDEFINED) {
```

Comment:
1. Please cache the repeating `residues[i]._type` as `residueType` for the sake of speed and size.
2. `ResidueType.Flags` are **bit flags** for various combinations! You cannot just compare or assign them (as below).
3. I don't think `ResidueType.Flags.UNDEFINED` will help.

Another my concern is that you are assigning the type of all residues in a chain, even ligands. Please check 4NRE.pdb for FE2, C8E, SO4 residues. They are neither PROTEIN nor NUCLEIC, and other algorithms may fail after such an assignment.

We need a different solution, e.g.
- do not guess residue types now,
- introduce `PROTEIN_CHAIN` and `NUCLEIC_CHAIN` if you really must.

Let's discuss.

Reply:
The idea about ResidueType.Flags.UNDEFINED is that before my changes if the residue is neither nucleic neither protein neither water then its residues[i]._type is equal to 0x0000 - for me that is the magic number, I decided to call it. 

The algorithm handles two principle different cases: if the residue is nucleic or not. So to all residues that aren't nucleic, I assign type "Amino". Yes, it seems, that there are residues that are not either nucleic either amino, but the principle difference: are they nucleic or not.
So, I see the decision to use enum like NUCLEIC, NOT_NUCLEIC.  
What do you think about this?
Interest fact: water residues need to be handled as NUCLEIC.

Reply:
Finally, we decided to choose a slightly different approach.